ruby/lib/ThreatExchange.rb

```diff
@@ -78,72 +53,62 @@ def threat_indicators(filter={})
       return data
     end
 
-    def indicator_pq(filter={})
-      begin
-        response = RestClient.get "#{@baseurl}/#{filter[:id]}/",
-        { params: { access_token: @access_token, fields: filter[:fields] } }
-        result = JSON.parse(response)
-        return result
-      rescue => e
-        puts e.inspect
-      end
+    def indicator_pq(params={})
+      get "#{@baseurl}/#{params[:id]}/", params
     end
 
-    def members()
-      begin
-        response = RestClient.get "#{@baseurl}/threat_exchange_members/",
-        { params: { access_token: @access_token } }
-        result = JSON.parse(response)
-        return result['data']
-      rescue => e
-        e.response
-      end
+    def members(params={})
+      get "#{@baseurl}/threat_exchange_members/", params
     end
 
-    def new_relation(data={})
-      data[:access_token] = @access_token
-      id = data.delete(:id)
-      begin
-        response = RestClient.post "#{@baseurl}/#{id}/related", data
-      rescue => e
-        e.inspect
-      end
+    def new_relation(params={})
+      id = params.delete(:id)
+
+      post "#{@baseurl}/#{id}/related", params
     end
 
-    def remove_relation(data={})
-      data[:access_token] = @access_token
-      id = data.delete(:id)
-      begin
-        response = RestClient.delete "#{@baseurl}/#{id}/related/",
-        { params: data }
-      rescue => e
-        e.response
-      end
+    def remove_relation(params={})
+      id = params.delete(:id)
+
+      delete "#{@baseurl}/#{id}/related/", params
     end
 
-    def new_ioc(data={})
-      data[:access_token] = @access_token
-      if data.has_key?(:privacy_type)
-        begin
-          response = RestClient.post "#{@baseurl}/threat_indicators", data
-          return response
-        rescue => e
-          e.inspect
-        end
+    def new_ioc(params={})
+      if params.has_key?(:privacy_type)
+        post "#{@baseurl}/threat_indicators", params
       else
         puts "You must set a privacy_type in your query"
```

Comment:
I understand this was here before, but would be great if this could be changed to raise an exception instead (e.g. `ArgumentError`).